analytics/http/filters.go

```diff
@@ -0,0 +1,213 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return func(event *analytics.CookieSyncObject) bool {
+		if event == nil || feature.SampleRate <= 0 || randomGenerator.GenerateFloat64() > feature.SampleRate {
+			return false
+		}
+
+		if filterProgram != nil {
+			output, err := expr.Run(filterProgram, event)
+			if err != nil {
+				glog.Errorf("[HttpAnalytics] Error filter cookie sync: %v", err)
+				return false
+			}
+			return output.(bool)
+		}
+
+		return true
+	}, nil
+}
+
+func createNotificationFilter(
+	feature config.AnalyticsFeature,
+	randomGenerator randomutil.RandomGenerator,
+) (notificationFilter, error) {
+	var filterProgram *vm.Program
+	var err error
+
+	if feature.Filter != "" {
+		filterProgram, err = expr.Compile(feature.Filter, expr.Env(analytics.NotificationEvent{}), expr.AsBool())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return func(event *analytics.NotificationEvent) bool {
+		if event == nil || feature.SampleRate <= 0 || randomGenerator.GenerateFloat64() > feature.SampleRate {
+			return false
+		}
+
+		if filterProgram != nil {
+			output, err := expr.Run(filterProgram, event)
+			if err != nil {
+				glog.Errorf("[HttpAnalytics] Error filter notification: %v", err)
+				return false
+			}
+			return output.(bool)
+		}
+
+		return true
+	}, nil
+}
+
+func createSetUIDFilter(
+	feature config.AnalyticsFeature,
+	randomGenerator randomutil.RandomGenerator,
+) (setUIDFilter, error) {
+	var filterProgram *vm.Program
+	var err error
+
+	if feature.Filter != "" {
+		filterProgram, err = expr.Compile(feature.Filter, expr.Env(analytics.SetUIDObject{}), expr.AsBool())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return func(event *analytics.SetUIDObject) bool {
+		if event == nil || feature.SampleRate <= 0 || randomGenerator.GenerateFloat64() > feature.SampleRate {
+			return false
+		}
+
+		if filterProgram != nil {
+			output, err := expr.Run(filterProgram, event)
+			if err != nil {
+				glog.Errorf("[HttpAnalytics] Error filter setUID: %v", err)
+				return false
+			}
+			return output.(bool)
+		}
+
+		return true
+	}, nil
+}
+
+func createVideoFilter(
+	feature config.AnalyticsFeature,
+	randomGenerator randomutil.RandomGenerator,
+) (videoFilter, error) {
+	var filterProgram *vm.Program
+	var err error
+
+	if feature.Filter != "" {
+		filterProgram, err = expr.Compile(feature.Filter, expr.Env(analytics.VideoObject{}), expr.AsBool())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return func(event *analytics.VideoObject) bool {
+		if event == nil || feature.SampleRate <= 0 || randomGenerator.GenerateFloat64() > feature.SampleRate {
+			return false
+		}
+
+		if filterProgram != nil {
+			output, err := expr.Run(filterProgram, event)
+			if err != nil {
+				glog.Errorf("[HttpAnalytics] Error filter video: %v", err)
+				return false
+			}
+			return output.(bool)
+		}
+
+		return true
+	}, nil
+}
```

Comment:
Yeah I like the idea of consolidating this code. 

For my own understanding, I haven't worked with generics that much, how would they call this new function `createFilter()` instead of `createAuctionFilter()` for example?